.circleci/config.yml

```diff
@@ -41,51 +52,121 @@ jobs:
       - restore_cache:
           name: Restore Rust cache
           keys:
-            - v5-cargo-pr-tests-{{ checksum "Cargo.toml" }}
-
-      - run:
-          name: Install Rust toolchain 1.72 (with clippy and rustfmt)
-          command: |
-            rustup toolchain install 1.72-x86_64-unknown-linux-gnu
-            rustup component add clippy --toolchain 1.72-x86_64-unknown-linux-gnu
-            rustup component add rustfmt --toolchain 1.72-x86_64-unknown-linux-gnu
-
-      - run:
-          name: Install nightly
-          command: rustup toolchain install nightly-2023-01-03-x86_64-unknown-linux-gnu
-
-      - run:
-          name: Install riscv target
-          command: rustup target add riscv32imac-unknown-none-elf --toolchain nightly-2023-01-03-x86_64-unknown-linux-gnu
-
-      - run:
-          name: Install stdlib
-          command: rustup component add rust-src --toolchain nightly-2023-01-03-x86_64-unknown-linux-gnu
+            - v5-cargo-pr-tests-{{ .Branch }}
```

Comment:
I'm not sure I agree with this. The cache from `main` can also be useful.

Reply:
Yes, as a fallback we should use the cache from `main`. The problem is that it should be somewhat recent. I couldn't find a good way to do this yet.

Reply:
I think you can add an extra line with another cache key that is checked if the first one fails. We can actively invalidate it once in a while